core/commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/Arrays.java

```diff
@@ -0,0 +1,175 @@
+import com.google.common.collect.ObjectArrays;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Arrays.copyOf;
+
+/**
+ * Utility methods to operate with arrays.
+ *
+ * @author Vlad Zhukovskyi
+ * @since 4.3.0
+ */
+@Beta
+public class Arrays {
+
+    /**
+     * Checks where given {@code array} is {@code null} or empty.
+     *
+     * @param array
+     *         the array to check
+     * @param <T>
+     *         any type of the given array
+     * @return {@code true} if given array is null or empty, otherwise {@code false}
+     * @since 4.3.0
+     */
+    public static <T> boolean isNullOrEmpty(T[] array) {
+        return array == null || array.length == 0;
+    }
+
+    /**
+     * Adds the given {@code element} to the tail of given {@code array}.
+     *
+     * @param array
+     *         the array to which {@code element} should be inserted
+     * @param element
+     *         {@code element} to insert
+     * @param <T>
+     *         type of given {@code array}
+     * @return the copy of given {@code array} with added {@code element}
+     * @throws IllegalArgumentException
+     *         in case if given {@code arrays} is null
+     * @since 4.3.0
+     */
+    public static <T> T[] add(T[] array, T element) {
+        checkArgument(array != null, "Input array is null");
+
+        final int index = array.length;
+
+        array = java.util.Arrays.copyOf(array, index + 1);
+        array[index] = element;
+
+        return array;
+    }
+
+    /**
+     * Checks if given {@code element} exists in {@code array}.
+     *
+     * @param array
+     *         arrays to check
+     * @param element
+     *         element to check for existence
+     * @param <T>
+     *         type of given {@code array}
+     * @return {@code true} if input {@code element} exists in array, otherwise {@code false}
+     * @throws IllegalArgumentException
+     *         in case if given {@code array} is null
+     * @since 4.3.0
+     */
+    public static <T> boolean contains(T[] array, T element) {
+        return indexOf(array, element) != -1;
+    }
+
+    /**
+     * Returns the index of the first occurrence of the element {@code o} in given {@code array},
+     * or -1 if there is no such element exists.
+     *
+     * @param array
+     *         input array
+     * @param o
+     *         element to search for
+     * @param <T>
+     *         type of given {@code array}
+     * @return the index of the first occurrence of the specified element in this {@code array}, otherwise -1
+     * @throws IllegalArgumentException
+     *         in case if given {@code array} is null
+     * @since 4.3.0
+     */
+    public static <T> int indexOf(T[] array, T o) {
+        checkArgument(array != null, "Input array is null");
+
+        for (int index = 0; index < array.length; index++) {
+            if (array[index] == o) {
+                return index;
```

Comment:
Comparison has changed to use with guava method `com.google.common.base.Objects#equal`